tests/integration/schema/migrations/query/with_p2p_schema_branch_test.go

```diff
@@ -0,0 +1,138 @@
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package query
+
+import (
+	"testing"
+
+	"github.com/lens-vm/lens/host-go/config/model"
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+	"github.com/sourcenetwork/defradb/tests/lenses"
+)
+
+func TestSchemaMigrationQueryWithP2PReplicatedDocOnOtherSchemaBranch(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.RandomNetworkingConfig(),
+			testUtils.RandomNetworkingConfig(),
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+						verified: Boolean
+					}
+				`,
+			},
+			testUtils.SchemaPatch{
+				// Patch first node only
+				NodeID: immutable.Some(0),
+				Patch: `
+					[
+						{ "op": "add", "path": "/Users/Fields/-", "value": {"Name": "email", "Kind": "String"} }
+					]
+				`,
+			},
+			testUtils.ConfigureMigration{
+				// Register the migration on both nodes.
+				LensConfig: client.LensConfig{
+					SourceSchemaVersionID:      "bafkreihax57fohcdupqr2l4heoqxdsiggjfeaubr44tgrz4xqdgvnid4xy",
+					DestinationSchemaVersionID: "bafkreifer354qmdrwdtae5n3k7sbl2oauis3mz24fk46p3otub7ojznobq",
+					Lens: model.Lens{
+						Lenses: []model.LensModule{
+							{
+								Path: lenses.SetDefaultModulePath,
+								Arguments: map[string]any{
+									"dst":   "name",
+									"value": "Fred",
+								},
+							},
+						},
+					},
+				},
+			},
+			testUtils.SchemaPatch{
+				// Patch second node with different patch
+				NodeID: immutable.Some(1),
+				Patch: `
+					[
+						{ "op": "add", "path": "/Users/Fields/-", "value": {"Name": "phone", "Kind": 11} }
+					]
+				`,
+				Lens: immutable.Some(model.Lens{
+					Lenses: []model.LensModule{
+						{
+							Path: lenses.SetDefaultModulePath,
+							Arguments: map[string]any{
+								"dst":   "phone",
+								"value": "1234567890",
+							},
+						},
+					},
+				}),
+			},
+			testUtils.ConfigureReplicator{
+				SourceNodeID: 0,
+				TargetNodeID: 1,
+			},
+			testUtils.CreateDoc{
+				// Create John on the first (source) node only, and allow the value to sync
+				NodeID: immutable.Some(0),
+				Doc: `{
+					"name": "John",
+					"verified": true
+				}`,
+			},
+			testUtils.WaitForSync{},
+			testUtils.Request{
+				// Node 0 should yield results as they were defined, as the newer schema version is
+				// unknown to this node.
+				NodeID: immutable.Some(0),
+				Request: `query {
+					Users {
+						name
+					}
+				}`,
+				Results: []map[string]any{
+					{
+						"name":     "John",
+						"verified": true,
+					},
+				},
```

Comment:
thougth: This seems to document a bug in our test system. Why is `verified` returned when only `name` is requested. Also, we should be able to return the `email` field here.

Reply:
> This seems to document a bug in our test system.

True, I forgot to add that to the query
- [ ] Fix test query

> Why is verified returned when only name is requested.

It wont be returned, I think it is far more likely that our test system is not asserting that all expected fields are returned.

> Also, we should be able to return the email field here.

Of course, but it is not relevant to this test and would just be noise here .

Reply:
> Of course, but it is not relevant to this test and would just be noise here .

That is subjective. I find that since a patch was applied to the schema, I would like to see that the field is available to query. Otherwise I'm not sure why just the name was queried. One could argue that the comment above is misleading as it's not true that the newer version is unknown to this node. It is in fact on a newer schema version. Technically just as new to the node as the other version is to the other node.

Reply:
> I would like to see that the field is available to query.

There are other tests that test this, without complicating things with Lens transforms.

> One could argue that the comment above is misleading as it's not true that the newer version is unknown to this node

I don't understand you here. schema version 3 is unknown, it doesnt exist on this node, only the migration exists not the definition.